include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -1067,19 +1126,68 @@ namespace CUDAWrappers
     if (src.get_partitioner().get() == partitioner.get() &&
         dst.get_partitioner().get() == partitioner.get())
       {
-        src.update_ghost_values();
+        if (overlap_communication_computation)
```

Comment:
Could you add a reference to https://github.com/dealii/dealii/blob/f01c9189968ad7f7d9cff0db077a9be5c77527e8/source/matrix_free/task_info.cc#L580-L620?